utils/fileutils.go

```diff
@@ -585,42 +569,38 @@ func GetFileDetails(filePath string, includeChecksums bool) (details *FileDetail
 	if includeChecksums {
 		details.Checksum, err = calcChecksumDetails(filePath)
 		if err != nil {
-			return details, err
+			return
 		}
 	} else {
 		details.Checksum = entities.Checksum{}
 	}
 
 	file, err := os.Open(filePath)
 	defer func() {
-		e := file.Close()
-		if err == nil {
-			err = e
-		}
+		err = errors.Join(err, file.Close())
 	}()
 	if err != nil {
-		return nil, err
+		return
 	}
 	fileInfo, err := file.Stat()
 	if err != nil {
-		return nil, err
+		return
 	}
 	details.Size = fileInfo.Size()
-	return details, nil
+	return
 }
 
 func calcChecksumDetails(filePath string) (checksum entities.Checksum, err error) {
 	file, err := os.Open(filePath)
-	defer func() {
-		e := file.Close()
-		if err == nil {
-			err = e
-		}
-	}()
 	if err != nil {
-		return entities.Checksum{}, err
+		return
 	}
-	return calcChecksumDetailsFromReader(file)
+	defer func() {
+		err = errors.Join(err, file.Close())
+	}()
+
+	checksum, err = calcChecksumDetailsFromReader(file)
+	return
 }
 
 func calcChecksumDetailsFromReader(reader io.Reader) (entities.Checksum, error) {
```

Comment:
Not that "calcChecksumDetails" is short and beautiful, I think we can remove "calcChecksumDetailsFromReader" and pour its content into "calcChecksumDetails".